hnn_core/drives.py

```diff
@@ -267,7 +260,8 @@ def _drive_cell_event_times(drive_type, drive_conn, dynamics,
     # than desired
     # values MUST be sorted for VecStim()!
     if len(event_times) > 0:
-        event_times = event_times[event_times > 0]
+        event_times = event_times[np.logical_and(event_times > 0,
+                                                 event_times <= tstop)]
```

Comment:
this feels like a hack. Why not just raise an error?

Reply:
Why would you want to raise an error? The goal is to remove spike events that extend past `tstop`. If you want the input spike train to match the output spike train exactly, any input spikes that extend past the end of the simulation need to cropped (for instance, imagine an evoked drive added at `t = 165` when `tstop = 170`). An input like this already exists in `test_network.py` but didn't flag our attention because it was silently ignored due to our previous convention that removed drive trains with zero synaptic weight.

Reply:
*as noted [below](https://github.com/jonescompneurolab/hnn-core/pull/383#discussion_r671776755), the `tstop` I'm specifically trying to access here is the stop time of the simulation which is not necessarily the stop time of the (bursty or poisson) drive.

Reply:
right, I guess I find the whole truncation business weird. If you have an evoked drive added at t = 165 with a mu of 25, if you truncate at 170, what you get is not a mu of 25 ... that's why I suggested raising an error

Reply:
You're right that you don't get a mu that is observable post-hoc as occurring at 165 ms, however, you get a drive that was generated a priori at 165 ms. In my mind, this is the whole point of using a biophysically principled neural simulator. Here, we simulate a reduced representation of a true biological process. If that process gets cropped in time (as is inevitably done in a real experiment), we can observe the spike time histogram (and subsequent model response) while also knowing the true generative probability distribution.

Conversely, we'd have to raise an error any time at least one of the 270 gaussian samples happens to fall outside the time window defined by the duration of the simulation.  Even for a gaussian distribution reasonably distanced from `tstop`, such a scenario can be fairly probable.

Reply:
Besides, we are already functionally cropping these time samples from the model response by setting a simulation stop time. All these lines of code do is allow us to match the spike events specified in `Network.external_drives` to those in `Network.cell_response`.

Reply:
makes sense!